.github/workflows/upstreams-tests.yml

```diff
@@ -0,0 +1,55 @@
+name: upstream-tests
+on:
+  push:
+    branches:
+      - '**'
+  pull_request:
+    paths:
+      - '**'
+      - '!doc/**'
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - run: make build
+  unit:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - run: make unit
+      - run: sed -i'' "s:^github.com/$GITHUB_REPOSITORY/::" coverage.out
+      - uses: codecov/codecov-action@v1
+        with:
+          file: coverage.out
+          fail_ci_if_error: false
+  run-e2e-minikube:
+    runs-on: ubuntu-latest
```

Comment:
we should add e2e tests for macos-latest and windows-latest too

Reply:
that's definitely a good idea especially for `opm`
but I think first we should work on getting the e2e tests to pass 😄 
then add additional os's as a followup